ping/rust/src/lib.rs

```diff
@@ -28,33 +29,39 @@ pub trait PingSwarm: Sized {
 }
 
 pub async fn run_ping<S>(mut swarm: S) -> Result<()>
-where
-    S: PingSwarm,
+    where
+        S: PingSwarm,
 {
-    info!("Running ping test: {}", swarm.local_peer_id());
-
     env_logger::Builder::from_env(Env::default().default_filter_or("info")).init();
 
-    let client = testground::client::Client::new_and_init().await.unwrap();
+    info!("Running ping test: {}", swarm.local_peer_id());
 
-    let local_addr = match if_addrs::get_if_addrs()
-        .unwrap()
+    let client = testground::client::Client::new_and_init()
+        .await
+        .expect("Unable to init testground cient.");
+    let transport = transport_param(&client);
+    let local_ip_addr = match if_addrs::get_if_addrs()?
         .into_iter()
         .find(|iface| iface.name == "eth1")
-        .unwrap()
+        .ok_or_else(|| io::Error::new(io::ErrorKind::Other, "Can't find iface eth1"))?
```

Comment:
```suggestion
        .context("Can't find iface eth1"))?
```

We are using anyhow so this would work too!

Reply:
Thanks for the suggestion!  This has been updated.